include/votca/xtp/fileadapter.h

```diff
@@ -0,0 +1,211 @@
+        assert(from_csg_top.MoleculeCount()==1 && "It is only possible to convert a csg topology object to a atom container when there is a single molecule in the topology object, otherwise it is not clear which molecule should be converted"); 
+
+        convertCSGTopologyToXTPContainer(from_csg_top,to_cont,0);
+      }
+
+    template<class CSG_Topology_T, class Container_T>  
+      inline void FileAdapter::convertCSGTopologyToXTPContainer(CSG_Topology_T & from_csg_top, Container_T & to_cont,const int & csg_molecule_id){
+        assert(from_csg_top.MoleculeExist(csg_molecule_id) && "Cannot convert csg molecule object with provided id to a atom container, no csg molecule with the specified id exists"); 
+
+        typename CSG_Topology_T::container_t csg_cont = from_csg_top.getMolecule(csg_molecule_id);
+        tools::StructureParameters params = csg_cont.getParameters();
+        /// Set the molecule id and type equal to the atom container id and type
+        int mol_id = params.get<int>(tools::StructureParameter::MoleculeId);
+        std::string molecule_type = params.get<std::string>(tools::StructureParameter::MoleculeType);
+        std::unordered_map<int,std::vector<tools::StructureParameters>> mol_id_and_bead_params;
+
+        std::vector<int> bead_ids = csg_cont.getBeadIds();
+        for( const int & bead_id : bead_ids ){
+          typename CSG_Topology_T::bead_t * bead = csg_cont.getBead(bead_id); 
+          tools::StructureParameters params_bead = bead->getParameters();
+          mol_id_and_bead_params[mol_id].push_back(params_bead);
+        }
+
+        to_cont = Container_T(molecule_type,mol_id);
+
+        for( tools::StructureParameters bead_params : mol_id_and_bead_params[mol_id]){ 
+          Eigen::VectorXd pos = bead_params.get<Eigen::VectorXd>(tools::StructureParameter::CSG_Position);
+          pos*= converter_.convert(CSG_Topology_T::distance_unit,Container_T::distance_unit);
+          bead_params.set(tools::StructureParameter::XTP_Position,pos);
+          typename Container_T::bead_t bead(bead_params);
+          to_cont.push_back(bead);
+        }
+
+      }
+
+    template<class CSG_Topology_T, class XTP_Topology_T>  
+      inline void FileAdapter::convertCSGTopologyToXTPTopology(CSG_Topology_T & from_csg_top, XTP_Topology_T & to_xtp_top){
+
+        Eigen::Matrix3d box = from_csg_top.getBox() * 
+          converter_.convert(CSG_Topology_T::distance_unit,XTP_Topology_T::distance_unit); 
+        to_xtp_top.setBox(box);
+
+        to_xtp_top.setStep(from_csg_top.getStep());
+        to_xtp_top.setTime(from_csg_top.getTime() * 
+            converter_.convert(CSG_Topology_T::time_unit,XTP_Topology_T::time_unit));
+
+        std::vector<std::pair<int,std::string>> id_and_types;
+        std::unordered_map<int,std::vector<tools::StructureParameters>> mol_id_and_bead_params;
+        /// We must collect the molecules and beads in order because the 
+        /// order that they are created in xtp is important. 
```

Comment:
why? I do not understand the comment

Reply:
xtp assigns ids to the atoms in the order they are created, so for the ids of the beads in csg to line up with the ids of atoms in xtp they first needed to be ordered.